packages/ariakit/src/combobox/__examples__/combobox-highlighted-text/test.tsx

```diff
@@ -0,0 +1,145 @@
+import { click, getByRole, getByText, press, render, type } from "ariakit-test";
+import { axe } from "jest-axe";
+import Example from ".";
+
+const getCombobox = () => getByRole("combobox");
+const getPopover = () => getByRole("listbox", { hidden: true });
+const getOption = (name: string) => getByRole("option", { name, exact: false });
+
+test("a11y", async () => {
+  const { container } = render(<Example />);
+  expect(await axe(container)).toHaveNoViolations();
+});
+
+test.only("show on click", async () => {
```

Comment:
```suggestion
test("show on click", async () => {
```

Reply:
Now the tests who activate the highlight are failing, it seems that "exact: false" from `getOption` is not enough..

On some older projects I've used a custom fuzzyMatch function:

```typescript
import {Matcher, MatcherFunction} from '@testing-library/react'

/**
 * function to create a MatcherFunction useful when you need to match
 * a complex html case, like: <p>I am a complex <b>string</b></p>
 *
 * Jest implentation example
 * expect(
 *       screen.getByText(fuzzyMatcher('I am a complex string')),
 * ).toBeVisible()
 */
export function fuzzyMatcher(
  matcher: Exclude<Matcher, MatcherFunction>,
): MatcherFunction {
  const matcherFunction: MatcherFunction = (_content, node) => {
    if (node == null) {
      return false
    }

    const hasText = (node: Element) => {
      if (matcher instanceof RegExp) {
        return node.textContent != null && matcher.test(node.textContent)
      }

      return node.textContent === matcher
    }
    const nodeHasText = hasText(node)
    // eslint-disable-next-line testing-library/no-node-access
    const childrenDontHaveText = Array.from(node?.children ?? []).every(
      (child) => !hasText(child),
    )
    return nodeHasText && childrenDontHaveText
  }

  return matcherFunction
}
``` 

I dont have the time now to test it, I'll try it later.

@diegohaz if you have a better idea, i'll be happy to hear it 😄

Reply:
That looks like a bug on `@testing-library/react` (or `@testing-library/dom`). The accessible names of the options shouldn't have spaces between `span`s.

Reply:
Fixed